<p>.Rmd script used to generate this page here:
https://github.com/ljcohen/ljcohen.github.io/blob/master/analyses/Dasen/scripts/Dasen_RNAseq_report_brachial_thoracic_overlap.Rmd</p>

<p>Filenames for brachial-level transcripts are here:</p>

<p><code>
## [1] "BR-A-Control_counts.txt" "BR-A-Mutant_counts.txt" 
## [3] "BR-B-Control_counts.txt" "BR-B-Mutant_counts.txt" 
## [5] "BR-C-Control_counts.txt" "BR-C-Mutant_counts.txt"
</code></p>

<p>Filenames for thoracic-level transcripts are here:</p>

<p><code>
## [1] "TH-A-Control_counts.txt" "TH-A-Mutant_counts.txt" 
## [3] "TH-B-Control_counts.txt" "TH-B-Mutant_counts.txt" 
## [5] "TH-C-Control_counts.txt" "TH-C-Mutant_counts.txt"
</code></p>

<p>The size of the table with all brachial transcripts is:</p>

<p><code>
## [1] 12701    15
</code></p>

<p>The size of the brachial table with only significant transcripts, padj&lt;0.05 is:</p>

<p><code>
## [1] 64 15
</code></p>

<p>The size of the table with all thoracic transcripts is:</p>

<p><code>
## [1] 15557    15
</code></p>

<p>The size of the thoracic table with only significant transcripts, padj&lt;0.05 is:</p>

<p><code>
## [1] 129  15
</code></p>

<h1 id="venn-diagram">Venn Diagram</h1>

<p><img src="figure/unnamed-chunk-7-1.png" alt="plot of chunk unnamed-chunk-7" /></p>

<h1 id="heatmap">Heatmap</h1>

<p><code>
## Error in hclust(as.dist(1 - cor(t(d), method = "pearson")), method = "complete"): must have n &gt;= 2 objects to cluster
</code></p>

<p><code>
## Error in nrow(tree$merge): object 'hr' not found
</code></p>

<p><code>
## Error in unique(mycl): error in evaluating the argument 'x' in selecting a method for function 'unique': Error: object 'mycl' not found
</code></p>

<p><code>
## Error in eval(expr, envir, enclos): object 'clusterCols' not found
</code></p>

<p><code>
## Error in as.dendrogram(hr): object 'hr' not found
</code></p>

<p>File:</p>

<p><code>r
write.csv(Brachial_Thoracic_results,"Dasen_Brachial_Thoracic_merged_padj0.05.csv")
</code></p>

<h1 id="contact">Contact</h1>

<p>Lisa Cohen, PhD student in MCIP, UC Davis   <br />
Phone: 321-427-9335     <br />
Email: ljcohen@ucdavis.edu</p>

<h3 id="sequencing-and-original-bioinformatics-analysis-by">Sequencing and original bioinformatics analysis by:</h3>

<p>NYU Langone Medical Center <br />
Bioinformatics Core, Genome Technology Center, OCS <br />
Email: Genomics@nyumc.org       <br />
Phone: 646-501-2834 <br />
http://ocs.med.nyu.edu/bioinformatics-core<br />
http://ocs.med.nyu.edu/genome-technology-center</p>

<p>Dr. Steven Shen, Associate Professor    <br />
Email: ShuQuan.Shen@nyumc.org<br />
Phone: 212-263-2234         <br />
Genome Technology Center, Office of Collaborative Sciences
and Center for Health Informatics and Bioinformatics
NYU School of Medicine,
227 E30th Street, Room 748, 
New York, NY 10016</p>

<h1 id="references">References</h1>

<p>M. I. Love, W. Huber, S. Anders: Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.
Genome Biology 2014, 15:550. http://dx.doi.org/10.1186/s13059-014-0550-8</p>

<p>R-Bioconductor: http://www.bioconductor.org/</p>

<p>DESeq2: http://www.bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.pdf</p>
